buildpack/build.go

```diff
@@ -81,13 +82,57 @@ func (b *Descriptor) Build(bpPlan Plan, config BuildConfig) (BuildResult, error)
 		return BuildResult{}, err
 	}
 
+	if err := b.processLayerTypes(bpLayersDir, config.Out); err != nil {
+		return BuildResult{}, err
+	}
+
 	if err := b.setupEnv(config.Env, bpLayersDir); err != nil {
 		return BuildResult{}, err
 	}
 
 	return b.readOutputFiles(bpLayersDir, bpPlanPath, bpPlan, config.Out)
 }
 
+func renameLayerDirIfNeeded(layerMetadataFile layertypes.LayerMetadataFile, layerDir string) error {
+	// rename <layers>/<layer> to <layers>/<layer>.ignore if buildpack API >= 0.6 and all of the types flags are set to false
+	if !layerMetadataFile.Launch && !layerMetadataFile.Cache && !layerMetadataFile.Build {
+		if err := os.Rename(layerDir, layerDir+".ignore"); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (b *Descriptor) processLayerTypes(layersDir string, out io.Writer) error {
+	if api.MustParse(b.API).Compare(api.MustParse("0.6")) < 0 {
+		return eachDir(layersDir, b.API, func(path, buildpackAPI string) error {
+			_, msg, err := DecodeLayerMetadataFile(path+".toml", buildpackAPI)
+			if err != nil {
+				return err
+			}
+			if msg != "" {
+				if _, err = out.Write([]byte(msg)); err != nil {
```

Comment:
Now that you pointed it out, I think we are also doing something similar here: https://github.com/buildpacks/lifecycle/blob/11ca8ef99c1626fa34ea087e74993d8746c7cf61/buildpack/build.go#L291 and here: https://github.com/buildpacks/lifecycle/blob/11ca8ef99c1626fa34ea087e74993d8746c7cf61/builder.go#L79

We'll probably want to pass a logger to the builder (like we do with analyzer). We could probably also pass this logger to the encoder/decoder to avoid the weirdness mentioned here: https://github.com/buildpacks/lifecycle/pull/537#discussion_r587842939 and here: https://github.com/buildpacks/lifecycle/pull/537#discussion_r593278527

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/537/commits/f80c0ccd77f7f3c0564ce125570dd551fe44b19f.
I didn't pass the `Logger` to the `Decoder` for now since the interface is defined in the `buildpack` package and in order to prevent circular imports, we can't import it in the `v05` and `v06` packages.
I'm planning to refactor these files a bit in order to get rid of the `layertypes` package, so I might do it in one of the next commits.

Reply:
Just for the record - the commit https://github.com/buildpacks/lifecycle/pull/537/commits/f3807d339969c123835663f8f65132b17cd2a8f6 is also relevant.